core/src/main/scala/org/bitcoins/core/psbt/PSBT.scala

```diff
@@ -203,8 +204,12 @@ case class PSBT(
           inputMap.redeemScriptOpt.isDefined && WitnessScriptPubKey
             .isWitnessScriptPubKey(
               inputMap.redeemScriptOpt.get.redeemScript.asm)
+        val notBIP143Vulnerable =
```

Comment:
is this right? If i understand this, it applies to every script that exists in the bitcoin ecosystem no?

Reply:
It is for the following if statement, we check if it is a witness output, then we also need to check if it's V0. So if it's non-segwit or SegwitV0, we use full tx, otherwise we use just the output (segwitV1+)